src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -120,7 +120,7 @@ const getRootClient = (
   const endpoint = dataSourceAttr.endpoint;
   const cachedClient = getClientFromPool(endpoint);
   if (cachedClient) {
-    return cachedClient;
+    return cachedClient as Client;
```

Comment:
Yes, because the client pool is updated to have 2 types for both legacy and new client. Have to cast the type both here and `configure_legacy_client.ts` so  that typescript won't complain 
```
private cache?: LRUCache<string, Client | LegacyClient>;
```